app/src/main/java/com/epfl/drawyourpath/mainpage/fragments/CommunityFragment.kt

```diff
@@ -147,66 +144,64 @@ class CommunityFragment : Fragment(R.layout.fragment_community) {
         val menu = view.findViewById<NavigationView>(R.id.community_navigation_view).menu
         createTournamentButton(menu)
 
-        val weekly = menu.addSubMenu("Weekly tournament")
-        if (tournament.getWeeklyTournament() != null) {
-            createMenuItem(view, weekly, tournament.getWeeklyTournament()!!)
+        val your = menu.addSubMenu("Your tournament")
+        tournamentModel.yourTournament.observe(viewLifecycleOwner) { tournaments ->
+            your.clear()
+            tournaments.forEach { createMenuItem(view, your, it, true) }
         }
 
-        val your = menu.addSubMenu("Your tournament")
-        for (t in tournament.getYourTournaments("placeholder")) {
-            createMenuItem(view, your, t)
+        val soon = menu.addSubMenu("Starting soon")
+        tournamentModel.startingSoonTournament.observe(viewLifecycleOwner) { tournaments ->
+            soon.clear()
+            tournaments.forEach { createMenuItem(view, soon, it.first, it.second) }
         }
         val discover = menu.addSubMenu("Discover")
-        for (t in tournament.getDiscoverTournaments("placeholder")) {
-            createMenuItem(view, discover, t)
+        tournamentModel.discoverTournament.observe(viewLifecycleOwner) { tournaments ->
+            discover.clear()
+            tournaments.forEach { createMenuItem(view, discover, it, false) }
         }
     }
 
     private fun createTournamentButton(menu: Menu) {
         menu.add(getString(R.string.create_new_tournament))
             .setIcon(R.drawable.ic_add)
             .setOnMenuItemClickListener {
-                // Sends args for mock to tournament creation fragment, only used during testing
-                // When not testing, db and auth are null, so no args are passed to the
-                // TournamentCreationFragment.
-
-                // Get database and auth types from the activity's launching intent.
-                val db = activity?.intent?.getStringExtra("Database")
-                val auth = activity?.intent?.getStringExtra("Auth")
-                // Put the strings in the activity's intent as keys with values true in a bundle used
-                // by the TournamentCreationFragment to choose which database and auth to use.
-                val fragmentArgs = Bundle()
-                if (db != null) fragmentArgs.putBoolean(db, true)
-                if (auth != null) fragmentArgs.putBoolean(auth, true)
-                // Launches the fragment
-                replaceFragment<TournamentCreationFragment>(fragmentArgs)
+                replaceFragment<TournamentCreationFragment>()
                 true
             }
```

Comment:
Why do you return true at the end of this call ?

Reply:
the OnMenuItemClickListener must return a boolean.